test/e2e/pv_reclaimpolicy.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	vsphere "k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/pkg/volume/util/volumehelper"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = framework.KubeDescribe("persistentvolumereclaim", func() {
+	f := framework.NewDefaultFramework("persistentvolumereclaim")
+	var c clientset.Interface
+	var ns string
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(c, framework.TestContext.NodeSchedulableTimeout))
+	})
+
+	/*
+		This test verifies persistent volume should be deleted when reclaimPolicy on the PV is set to delete and
+		associated claim is deleted
+
+		Test Steps:
+		1. Create vmdk
+		2. Create PV Spec with volume path set to VMDK file created in Step-1, and PersistentVolumeReclaimPolicy is set to Delete
+		3. Create PVC with the storage request set to PV's storage capacity.
+		4. Wait for PV and PVC to bound.
+		5. Delete PVC
+		6. Verify PV is deleted automatically.
+	*/
+	framework.KubeDescribe("persistentvolumereclaim:delete", func() {
+		var (
+			volumePath    string
+			pv            *v1.PersistentVolume
+			pvc           *v1.PersistentVolumeClaim
+			volumeoptions vsphere.VolumeOptions
+		)
+
+		It("should delete persistent volume when reclaimPolicy set to delete and associated claim is deleted", func() {
+
+			By("creating vmdk")
```

Comment:
I'd like to see this start up code, shared by delete and retain `It` 's, moved to a higher-level `BeforeEach` or moved to a helper func.